cmd/root.go

```diff
@@ -116,6 +116,10 @@ func rootCmd(execs execs) (*cobra.Command, error) {
 			if dbPort != 0 {
 				dbconn.Port = dbPort
 			}
+			if dbconn.Port == 0 {
+				// default is 3306
+				dbconn.Port = defaultPort
+			}
```

Comment:
Why? What is wrong with explicitly setting the default to 3306 as we had it before?

Reply:
If you set the default value at [root.go#L173](https://github.com/databacker/mysql-backup/blob/624330f6fd386315788d096915da4f5033186e0b/cmd/root.go#L173) as before, then [root.go#L115](https://github.com/databacker/mysql-backup/blob/624330f6fd386315788d096915da4f5033186e0b/cmd/root.go#L115) will always get a non-0 value, which will cause the port configuration to always be changed to 3306 when I use the configuration file.

Reply:
I get it. This is the classic problem of, "is this because someone set it (and so should override the config file), or is it because it defaulted to that number (and so should use the config file)"?

I kept it like this, because I far prefer the help to say explicitly "default is 3306" rather than "default is 0" and then the help message says, "but I will treat that as 3306". Explicit > implicit.

The "is it set" problem can be fixed. I will address this specific one.

Reply:
Rebase on latest `master`, it should have resolved this issue now.

Reply:
I think you should test it locally first instead of rushing to submit, The code I submitted has been tested locally by me, and I have confirmed that it can fix the issue I mentioned.

Reply:
Right. That is why you removed L94  it in this PR